### PR TITLE
[FIX] hw_drivers: Avoid duplicates identifiers in devices

### PR DESCRIPTION
**Pull request by qle-odoo**
Before this commit the identifier integrate the BUS and port to identify USB devices
Now we check if there are duplicates devices before increment the identifier

For other loop we add identifier in the iot_device object

task: 1963500

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hw_drivers/controllers/driver.py [hunk 1]
```diff
@@ -316,8 +320,9 @@ def video_loop(self):
                 dev = v4l2.v4l2_capability()
                 ioctl(path, v4l2.VIDIOC_QUERYCAP, dev)
                 dev.interface = video
+                dev.identifier = dev.bus_info.decode('utf-8')
                 iot_device = IoTDevice(dev, 'video')
-                camera_devices[dev.bus_info.decode('utf-8')] = iot_device
+                camera_devices[dev.identifier] = iot_device
```
Comment (sle-odoo) [inline addons/hw_drivers/controllers/driver.py]: you don't have to increment here?

File: addons/hw_drivers/controllers/driver.py [hunk 2]
```diff
@@ -302,10 +302,14 @@ def send_alldevices(self):
     def usb_loop(self):
         usb_devices = {}
         devs = core.find(find_all=True)
+        cpt = 2
         for dev in devs:
-            path =  "usb_%04x:%04x_%03d_%03d_" % (dev.idVendor, dev.idProduct, dev.bus, dev.address)
+            dev.identifier =  "usb_%04x:%04x" % (dev.idVendor, dev.idProduct)
+            if dev.identifier in usb_devices:
+                dev.identifier += '_%s' % cpt
+                cpt += 1
```
Comment (sle-odoo) [inline addons/hw_drivers/controllers/driver.py]: shouldn't this counter be defined out of the usb_loop method? as a global variable? isn't it possible to have two times usb_xxx_1 if we go two times in usb_loop?

File: addons/hw_drivers/controllers/driver.py (hunk 1 above)
Comment (qle-odoo) [reply addons/hw_drivers/controllers/driver.py]: video interface have already a unique identifier by raspberry

File: addons/hw_drivers/controllers/driver.py (hunk 2 above)
Comment (qle-odoo) [reply addons/hw_drivers/controllers/driver.py]: we can't have 2 instance of odoo.
And we call only one time the manager in driver.py.
So normaly... we can't go two times in usb_loop. 
Your opinion ?

Comment (sle-odoo) [reply addons/hw_drivers/controllers/driver.py]: but it is called by `run` right? so every 0.3sec we go in this method and reset the counter to 2

File: addons/hw_drivers/controllers/driver.py [hunk 3]
```diff
@@ -302,10 +302,14 @@ def send_alldevices(self):
     def usb_loop(self):
```
Comment (sle-odoo) [inline addons/hw_drivers/controllers/driver.py]:
```
USB devices are identified by a combination of their `idVendor` and
`idProduct`. We can't be sure this combination in unique per equipment.
To still allow connecting multiple similar equipments, we complete the
identifier by a counter. The drawbacks are we can't be sure the equipments
will get the same identifiers after a reboot or a disconnect/reconnect.
```

add a docstring
```
Loops over the connected usb devices, assign them an identifier, instantiate
an `IoTDevice` and put it in `usb_devices`.
```

**Comment by sle-odoo** [comment]
add task id in commit message, usb_loop docstring should start by the goal of the method, then we're good to go

**Comment by qle-odoo** [comment]
@robodoo r+

**Pull request closed without merging**